.github/workflows/test.yml

```diff
@@ -12,10 +12,13 @@ jobs:
     timeout-minutes: 30
     steps:
     - uses: actions/checkout@v3
+    # We use the Determinate Systems installer for 2.18 because the
+    # Sequoia UID/GID changes have not yet been backported to the
+    # official installer for that version.
     - name: Install nix corresponding to latest stable channel
-      uses: cachix/install-nix-action@v23
+      uses: DeterminateSystems/nix-installer-action@main
```

Comment:
They recommend deploying the action directly from `main` in their documentation. I can adjust this if you think it's important, though I expect we're only talking a matter of days before a fixed 2.18 installer we can use is released and I'm not at a computer right now so it'll have to wait until tomorrow.

Reply:
FWIW the action downloads an unversioned installer anyway so there's no meaningful pinning to be done here.

Reply:
I'm fine with these changes being put in another PR, I think we will be able to pin both the nix-installer binary used and the Nix version installed using `source-url` and `nix-package-url`, the goal for me is that we have both installers being tested in the CI eventually

Reply:
The upstream installer only supports installing one version of Nix, AFAIK. Again this is just a temporary hack for a few days to ensure that we can test 2.18 at all. If we tried to use the upstream installer we could not activate the system. This was the only way I could make the tests not go red because of this PR.